libp2p/crypto/crypto.nim

```diff
@@ -95,6 +95,9 @@ const
   SupportedSchemesInt* = {int8(RSA), int8(Ed25519), int8(Secp256k1),
                           int8(ECDSA)}
 
+template orError(exp: untyped, err: CryptoError): untyped = 
+  ? (exp.mapErr do (_: auto) -> auto: err)
```

Comment:
hm, I'm not completely convinced hiding `?` in the template is a good idea, but let's go with it for now and see how it feels

Reply:
I removed that